test/integration/mecanum_drive_system.cc

```diff
@@ -0,0 +1,537 @@
+      for (; odomPoses.size() < 150 && sleep < maxSleep; ++sleep)
+      {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      }
+      ASSERT_NE(maxSleep, sleep);
+
+      // Odometry calculates the pose of a point that is located half way
+      // between the two wheels, not the origin of the model. For example,
+      // if the vehicle is commanded to rotate in place, the vehicle will
+      // rotate about the point half way between the two wheels, thus,
+      // the odometry position will remain zero.
+      // However, since the model origin is offset, the model position will
+      // change. To find the final pose of the model, we have to do the
+      // following similarity transformation
+      
+      math::Pose3d tOdomModel(0.554283,0,-0.325,0,0,0);
+      auto finalModelFramePose =
+          tOdomModel * odomPoses.back() * tOdomModel.Inverse();
+
+      // Odom for 3s
+      ASSERT_FALSE(odomPoses.empty());
+      EXPECT_EQ(150u, odomPoses.size());
+
+      auto expectedLowerPosition =
+          (forward ? poses[0].Pos() : poses[3999].Pos());
+      auto expectedGreaterPosition =
+          (forward ? poses[3999].Pos() : poses[0].Pos());
+
+      EXPECT_LT(expectedLowerPosition.X(), expectedGreaterPosition.X());
+      EXPECT_LT(expectedLowerPosition.Y(), expectedGreaterPosition.Y());
+      EXPECT_NEAR(expectedLowerPosition.Z(), expectedGreaterPosition.Z(), tol);
+      EXPECT_NEAR(poses[0].Rot().X(), poses[3999].Rot().X(), tol);
+      EXPECT_NEAR(poses[0].Rot().Y(), poses[3999].Rot().Y(), tol);
+      EXPECT_NEAR(poses[0].Rot().Z(), poses[3999].Rot().Z(), tol);
+
+      // The value from odometry will be close, but not exactly the ground truth
+      // pose of the robot model. This is partially due to throttling the
+      // odometry publisher, partially due to a frame difference between the
+      // odom frame and the model frame, and partially due to wheel slip.
+      EXPECT_NEAR(poses.back().Pos().X(), odomPoses.back().Pos().X(), 1e-2);
+      EXPECT_NEAR(poses.back().Pos().Y(), odomPoses.back().Pos().Y(), 1e-2);
+      // EXPECT_NEAR(poses.back().Pos().X(), finalModelFramePose.Pos().X(), 1e-2);
+      // EXPECT_NEAR(poses.back().Pos().Y(), finalModelFramePose.Pos().Y(), 1e-2);
+
+      // Verify velocity and acceleration boundaries.
+      // Moving time.
+      double t = 3.0;
+      double d = poses[3999].Pos().Distance(poses[0].Pos());
+      double v0 = 0;
+      double v = d / t;
+      double a = (v - v0) / t;
+
+      EXPECT_LT(v, 0.5);
+      EXPECT_LT(a, 1);
+      EXPECT_GT(v, -0.5);
+      EXPECT_GT(a, -1);
+
+    };
+
+  testCmdVel(true /*test forward movement*/);
+  testCmdVel(false /*test backward movement*/);
+  }
+};
+
+/////////////////////////////////////////////////
+// See: https://github.com/gazebosim/gz-sim/issues/1175
+// See: https://github.com/gazebosim/gz-sim/issues/630
+TEST_P(MecanumDriveTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(PublishCmd))
```

Comment:
I know this is copied from the DiffDrive test, but can we first try to enable it and see if it doesn't work on the other platforms?

Reply:
i don't quite get what do you mean, since my testing skills are weak. 
do you mean changing GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX to something that enables the tests on other OSs ?

Reply:
yeah I think something like: 
```suggestion
TEST_P(MecanumDriveTest, PublishCmd)
```

let's also not block on this